servers/shared-snowplow-consumer/src/snowplow/shareableList/shareableListEventHandler.ts

```diff
@@ -24,77 +23,67 @@ export class ShareableListEventHandler extends EventHandler {
+      context,
+    });
   }
 
   /**
    * Builds the Snowplow object_update event object. Extracts the event trigger type from the received payload.
    */
   private static generateShareableListEvent(
-    data: ShareableListEventPayloadSnowplow,
+    data: ShareableListEventBridgePayload,
   ): ObjectUpdate {
     return {
-      schema: shareableListEventSchema.objectUpdate,
-      data: {
-        trigger: data.eventType,
-        object: 'shareable_list',
-      },
+      trigger: data['detail-type'],
+      object: 'shareable_list',
     };
   }
 
   private static generateEventContext(
-    data: ShareableListEventPayloadSnowplow,
+    data: ShareableListEventBridgePayload,
   ): SelfDescribingJson[] {
-    return [ShareableListEventHandler.generateSnowplowShareableListEvent(data)];
+    return [
+      createShareableList(
+        ShareableListEventHandler.generateSnowplowShareableListEvent(
+          data.detail.shareableList,
+        ),
+      ) as unknown as SelfDescribingJson,
+    ];
   }
 
   /**
    * Static method to generate an object that maps properties received in the event payload object to the snowplow shareable_list object schema.
    */
   private static generateSnowplowShareableListEvent(
-    data: ShareableListEventPayloadSnowplow,
+    data: ShareableListEventBridgePayload['detail']['shareableList'],
   ): ShareableList {
-    const snowplowEvent = {
-      schema: shareableListEventSchema.shareable_list,
-      data: {
-        shareable_list_external_id:
-          data.shareable_list.shareable_list_external_id,
-        user_id: data.shareable_list.user_id
-          ? data.shareable_list.user_id
-          : undefined,
-        slug: data.shareable_list.slug,
-        title: data.shareable_list.title,
-        description: data.shareable_list.description
-          ? data.shareable_list.description
-          : undefined,
-        status: data.shareable_list.status,
-        list_item_note_visibility:
-          data.shareable_list.list_item_note_visibility,
-        moderation_status: data.shareable_list.moderation_status,
-        moderated_by: data.shareable_list.moderated_by
-          ? data.shareable_list.moderated_by
-          : undefined,
-        moderation_reason: data.shareable_list.moderation_reason
-          ? data.shareable_list.moderation_reason
-          : undefined,
-        moderation_details: data.shareable_list.moderation_details
-          ? data.shareable_list.moderation_details
-          : undefined,
-        restoration_reason: data.shareable_list.restoration_reason
-          ? data.shareable_list.restoration_reason
-          : undefined,
-        created_at: data.shareable_list.created_at,
-        updated_at: data.shareable_list.updated_at
-          ? data.shareable_list.updated_at
-          : undefined,
-      },
+    return {
+      shareable_list_external_id: data.shareable_list_external_id,
+      user_id: data.user_id ? data.user_id : undefined,
```

Comment:
since you're here already, it would be nice to clean this up.

either use the nullish coalesce operator (`data.user_id ?? undefined`), audit to understand whether we even need this nullish operator (e.g. is it possible for `user_id` to be null but not undefined), or build this with `map`/`reduce` since all the keys are identical

Reply:
fixed. idk why we didnt use `??`